src/amt/cim.ts

```diff
@@ -3,56 +3,251 @@
 * SPDX-License-Identifier: Apache-2.0
 **********************************************************************/
 
-import '../models/cim_models'
-import { WSManMessageCreator } from './wsman'
-
-export enum CIM_Methods {
-  GET = 'Get',
-  PULL = 'Pull',
-  ENUMERATE = 'Enumerate',
-  RELEASE = 'Release'
-}
-
-export enum CIM_Actions {
-  ENUMERATE = 'enumeration/Enumerate',
-  PULL = 'enumeration/Pull',
-  GET = 'transfer/Get'
-}
+import { Selector, WSManMessageCreator } from './wsman'
+import { CIM_Methods, CIM_Actions, CIM_Classes } from './enums/cim_enums'
 
 export class CIM {
-  wsmanMessageCreator: WSManMessageCreator
-  resourceUriBase: 'http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/'
-  commonCimRUri: 'http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/'
-  constructor () {
-    this.wsmanMessageCreator = new WSManMessageCreator()
+  wsmanMessageCreator: WSManMessageCreator = new WSManMessageCreator()
+  readonly resourceUriBase: string = 'http://schemas.dmtf.org/wbem/wscim/1/cim-schema/2/'
+  private enumerate (action: CIM_Actions, cimClass: CIM_Classes, messageId: String): String {
+    const header: String = this.wsmanMessageCreator.createHeader(action, `${this.resourceUriBase}${cimClass}`, messageId)
+    const body: String = this.wsmanMessageCreator.createBody(CIM_Methods.ENUMERATE)
+    const response: String = this.wsmanMessageCreator.createXml(header, body)
+    return response
+  }
+
+  private pull (action: CIM_Actions, cimClass: CIM_Classes, messageId: String, enumerationContext: String): String {
+    const header: String = this.wsmanMessageCreator.createHeader(action, `${this.resourceUriBase}${cimClass}`, messageId)
+    const body: String = this.wsmanMessageCreator.createBody(CIM_Methods.PULL, enumerationContext)
+    const response: String = this.wsmanMessageCreator.createXml(header, body)
+    return response
+  }
+
+  private get (action: CIM_Actions, cimClass: CIM_Classes, messageId: String): String {
+    const header: String = this.wsmanMessageCreator.createHeader(action, `${this.resourceUriBase}${cimClass}`, messageId)
+    const body: String = this.wsmanMessageCreator.createBody(CIM_Methods.GET)
+    const response: String = this.wsmanMessageCreator.createXml(header, body)
+    return response
   }
 
   cim_ServiceAvailableToElement = (method: CIM_Methods, messageId: String, enumerationContext?: String): String => {
-    let response: String, header: String, body: String
+    switch (method) {
+      case 'Pull':
```

Comment:
why not use the constant here in all the case conditions?

Reply:
agree.  There are a couple spots where the body is conditionally updated, but for everything else I'm using constants or just returning the response directly when it comes back from the called function.